core/new-gui/src/app/workspace/component/operator-panel/operator-label/operator-label.component.ts

```diff
@@ -16,21 +17,51 @@ import { OperatorSchema } from '../../../types/operator-schema.interface';
 })
 export class OperatorLabelComponent implements AfterViewInit {
 
+  @ViewChild('t') t: any; // t is the specific tooltip window for an operator
   @Input() operator?: OperatorSchema;
   public operatorLabelID: string;
+  private timer: any; // needed to add a delay to tooltip
+  private isHovering: boolean;
 
   constructor(
     private dragDropService: DragDropService
   ) {
     // generate a random ID for this DOM element
     this.operatorLabelID = 'texera-operator-label-' + uuid();
+
+    this.isHovering = false;
   }
 
   ngAfterViewInit() {
     if (! this.operator) {
       throw new Error('operator label component: operator is not specified');
     }
     this.dragDropService.registerOperatorLabelDrag(this.operatorLabelID, this.operator.operatorType);
+
+  }
+
+  // show the tooltip window after 1500ms
+  // var t means tooltip
+  public displayDescription(t: any): void {
+    this.isHovering = true;
+    const secondsCounter = interval(1500);
+
+    // filter(isHovering) ensures that the cursor is still inside the label
+    // when secontsCounter emites an observable
+    this.timer = secondsCounter.filter(val => this.isHovering).subscribe(val => (
+      this.t.open(),
+      // unsubscribe the timer after the first observable is received to
+      // ensure that the function is not called only once.
+      this.timer.unsubscribe()
```

Comment:
Why do we need to unsubscribe the timer here? Maybe we can just call something like 
`
    Observable.timer(1000).filter(val => this.isHovering).subscribe(val =>
        this.t.open()
    );
`
to make sure the mouse is currently hovering on top of one of the operator handle